src/mongo_odm/nvp.hpp

```diff
@@ -541,6 +630,11 @@ class nvp_base {
     }
 
     /* Array update operators */
+    template <typename U = no_opt_type, typename = std::enable_if_t<is_iterable_not_string_v<U>>>
+    constexpr dollar_operator_nvp<NvpT> update_first() const {
```

Comment:
`first_match` sounds good, since it returns a name-value pair (a "noun") that gets modified later when it's put into an update expression.